specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2023-04-01-preview/loadtestservice.json

```diff
@@ -0,0 +1,3301 @@
+          "type": "string"
+        },
+        "requestName": {
+          "description": "Request name for which the Pass fail criteria has to be applied ",
+          "type": "string"
+        },
+        "value": {
+          "format": "double",
+          "description": "The value to compare with the client metric. Allowed values - ‘error : [0.0 , 100.0] unit- % ’, response_time_ms and latency : any integer value unit- ms.",
+          "type": "number"
+        },
+        "action": {
+          "description": "Action taken after the threshold is met. Default is ‘continue’.",
+          "default": "continue",
+          "enum": [
+            "continue",
+            "stop"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PFAction",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "continue",
+                "description": "Test will continue to run even if pass fail metric criteria metric gets failed"
+              },
+              {
+                "value": "stop",
+                "description": "Test run will stop if pass fail criteria metric is not passed."
+              }
+            ]
+          }
+        },
+        "actualValue": {
+          "format": "double",
+          "description": "The actual value of the client metric for the test run.",
+          "type": "number",
+          "readOnly": true
+        },
+        "result": {
+          "description": "Outcome of the test run.",
+          "enum": [
+            "passed",
+            "undetermined",
+            "failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "PFResult",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "passed",
+                "description": "Given pass fail criteria metric has passed."
+              },
+              {
+                "value": "undetermined",
+                "description": "Given pass fail criteria metric couldn't determine."
+              },
+              {
+                "value": "failed",
+                "description": "Given pass fail criteria metric has failed."
+              }
+            ]
+          }
+        }
+      }
+    },
+    "ResourceMetric": {
+      "description": "Associated metric definition for particular metrics of the azure resource ( Refer : https://docs.microsoft.com/en-us/rest/api/monitor/metric-definitions/list#metricdefinition).",
+      "required": [
+        "aggregation",
+        "metricNamespace",
+        "name",
+        "resourceId",
+        "resourceType"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique name for metric.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resourceId": {
+          "description": "Azure resource id.",
+          "minLength": 1,
```

Comment:
`minLength` is useful _if_ it adds something to the constraints on valid values.  I doubt this is the case for `resourceId`, since the primary constraint for this field is that it contain a valid resource id, and this will always be larger than 1 character.

In short, I don't think this adds information here, so recommend you drop it.

Reply:
You've added `minLength` to several more properties in this schema. Please consider my comment above for whether these are really needed.

Reply:
removed minLength